dist/angular-date-picker.js

```diff
@@ -22,92 +22,121 @@ angular.module( "vokal.datePicker", [] )
     {
         "use strict";
 
-        var defaultFormat = "M/d/yyyy";
+        var defaultFormat = "M/D/YYYY";
 
-        function validateDate( date )
+        function validateDate( dateOrString )
         {
-            return !!date && angular.isFunction( date.getTime ) && !isNaN( date.getTime() );
-        }
-        function convertToDate( str )
-        {
-            return validateDate( str ) ? str : new Date( str );
+            return !!dateOrString && moment( new Date( dateOrString ) ).isValid();
         }
 
         return {
             restrict: "A",
-            scope: {},
+            scope: {
+                model: "=ngModel",
+                timezone: "="
+            },
             require: "ngModel",
             link: function ( scope, element, attrs, ngModelController )
             {
-                var localDate = new Date( new Date().toDateString() );
-                function filterOutput( date )
+                if( scope.timezone && !moment.tz )
+                {
+                    console.warn( "Trying to use moment-timezone without including the script." );
```

Comment:
Ouch on size. Even though it is optional, it is kind of all over the place. Would it be better to just make separate directives that explicitly do or do not have that dependency? Optional deps. are a bit of an anti-pattern.